<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Network Routing Simulator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOn</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustIgnored</enum>
      </property>
      <property name="renderHints">
       <set>QPainter::RenderHint::Antialiasing|QPainter::RenderHint::TextAntialiasing</set>
      </property>
      <property name="dragMode">
       <enum>QGraphicsView::DragMode::ScrollHandDrag</enum>
      </property>
      <property name="transformationAnchor">
       <enum>QGraphicsView::ViewportAnchor::AnchorUnderMouse</enum>
      </property>
      <property name="resizeAnchor">
       <enum>QGraphicsView::ViewportAnchor::AnchorViewCenter</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuNode">
    <property name="title">
     <string>Network</string>
    </property>
    <widget class="QMenu" name="menuAdd">
     <property name="title">
      <string>Add</string>
     </property>
     <addaction name="actionAddNode"/>
     <addaction name="actionAddConnection"/>
    </widget>
    <widget class="QMenu" name="menuDelete">
     <property name="title">
      <string>Delete</string>
     </property>
     <addaction name="actionDelete_Connection"/>
     <addaction name="actionDelete_Node"/>
    </widget>
    <widget class="QMenu" name="menuGenerate_topology">
     <property name="title">
      <string>Generate topology</string>
     </property>
     <addaction name="actionDuplex"/>
     <addaction name="actionHalf_Duplex"/>
     <addaction name="actionRandom"/>
    </widget>
    <widget class="QMenu" name="menuConvert_topology">
     <property name="title">
      <string>Convert topology</string>
     </property>
     <addaction name="actionConvert_to_duplex"/>
     <addaction name="actionConvert_to_half_duplex"/>
     <addaction name="actionConvert_to_random"/>
    </widget>
    <addaction name="menuAdd"/>
    <addaction name="menuDelete"/>
    <addaction name="actionClear"/>
    <addaction name="actionOpen"/>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_Out"/>
    <addaction name="menuGenerate_topology"/>
    <addaction name="actionArrange_Graph"/>
    <addaction name="actionSend_Message"/>
    <addaction name="menuConvert_topology"/>
   </widget>
   <widget class="QMenu" name="menuRouting">
    <property name="title">
     <string>Routing</string>
    </property>
    <addaction name="actionRouting_Start"/>
    <addaction name="actionRouting_Next"/>
    <addaction name="actionRouting_AutoRun"/>
    <addaction name="actionRouting_Reset"/>
   </widget>
   <widget class="QMenu" name="menuStart">
    <property name="title">
     <string>LSR</string>
    </property>
    <addaction name="actionLSR_Start"/>
    <addaction name="actionLSR_Next_Step"/>
    <addaction name="actionLSR_Run"/>
    <addaction name="actionLSR_Reset_2"/>
   </widget>
   <widget class="QMenu" name="menuMessage">
    <property name="title">
     <string>Message</string>
    </property>
    <addaction name="actionSend_Message"/>
    <addaction name="actionView_Logs"/>
   </widget>
   <widget class="QMenu" name="menuTest">
    <property name="title">
     <string>Test</string>
    </property>
    <addaction name="actionTest_Same_Pck_different_Msg_Size"/>
    <addaction name="actionTest_dif_Pck_same_Msg"/>
   </widget>
   <addaction name="menuNode"/>
   <addaction name="menuRouting"/>
   <addaction name="menuStart"/>
   <addaction name="menuMessage"/>
   <addaction name="menuTest"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionZoom_In"/>
   <addaction name="actionZoom_Out"/>
   <addaction name="actionAddNode"/>
   <addaction name="actionAddConnection"/>
   <addaction name="actionDelete_Connection"/>
   <addaction name="actionDelete_Node"/>
   <addaction name="actionSend_Message"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="nodeInspectorDock">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QWidget" name="widget" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Router ID</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="routerId">
          <property name="text">
           <string>Router</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QRadioButton" name="NodeON">
           <property name="text">
            <string>ON</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="NodeOFF">
           <property name="text">
            <string>OFF</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="labelTableTitle">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTableView" name="tableView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::ZoomIn"/>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::ZoomOut"/>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="actionAddNode">
   <property name="text">
    <string>Add Node</string>
   </property>
   <property name="toolTip">
    <string>Node</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionAddConnection">
   <property name="text">
    <string>Add Connection</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionDelete_Node">
   <property name="text">
    <string>Delete Node</string>
   </property>
  </action>
  <action name="actionDelete_Node_2">
   <property name="text">
    <string>Delete Node</string>
   </property>
  </action>
  <action name="actionDelete_Connection">
   <property name="text">
    <string>Delete Connection</string>
   </property>
   <property name="toolTip">
    <string>Delete Connection</string>
   </property>
  </action>
  <action name="actionShow_All">
   <property name="text">
    <string>Show All Connections</string>
   </property>
  </action>
  <action name="actionArrange_Graph">
   <property name="text">
    <string>Arrange Graph</string>
   </property>
  </action>
  <action name="actionRouting_Start">
   <property name="text">
    <string>Start SPF</string>
   </property>
  </action>
  <action name="actionRouting_Next">
   <property name="text">
    <string>Next Step</string>
   </property>
  </action>
  <action name="actionRouting_AutoRun">
   <property name="text">
    <string>Auto Run</string>
   </property>
  </action>
  <action name="actionRouting_Reset">
   <property name="text">
    <string>Reset Highlight</string>
   </property>
  </action>
  <action name="actionLSR_Next_Step">
   <property name="text">
    <string>LSR Next Step</string>
   </property>
  </action>
  <action name="actionLSR_Reset">
   <property name="text">
    <string>LSR Reset</string>
   </property>
  </action>
  <action name="actionLSR_Start">
   <property name="text">
    <string>LSR Start</string>
   </property>
  </action>
  <action name="actionLSR_Run">
   <property name="text">
    <string>LSR Run</string>
   </property>
  </action>
  <action name="actionLSR_Reset_2">
   <property name="text">
    <string>LSR Reset</string>
   </property>
  </action>
  <action name="actionSend_Message">
   <property name="text">
    <string>Send Message</string>
   </property>
  </action>
  <action name="actionView_Logs">
   <property name="text">
    <string>View Logs</string>
   </property>
  </action>
  <action name="actionTest_Same_Pck_different_Msg_Size">
   <property name="text">
    <string>Test Same Pck different Msg Size</string>
   </property>
  </action>
  <action name="actionTest_dif_Pck_same_Msg">
   <property name="text">
    <string>Test dif Pck same Msg</string>
   </property>
  </action>
  <action name="actionDuplex">
   <property name="text">
    <string>Duplex</string>
   </property>
  </action>
  <action name="actionHalf_Duplex">
   <property name="text">
    <string>Half-Duplex</string>
   </property>
  </action>
  <action name="actionRandom">
   <property name="text">
    <string>Random</string>
   </property>
  </action>
  <action name="actionConvert_to_duplex">
   <property name="text">
    <string>Convert to duplex</string>
   </property>
  </action>
  <action name="actionConvert_to_half_duplex">
   <property name="text">
    <string>Convert to half-duplex</string>
   </property>
  </action>
  <action name="actionConvert_to_random">
   <property name="text">
    <string>Convert to random</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
